src/serac/physics/solid_mechanics.hpp

```diff
@@ -896,13 +971,20 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
     // Build the dof array lookup tables
     displacement_.space().BuildDofToArrays();
 
+    // create contact mesh and compute forces, pressures, and Jacobians
+    if (contact_.haveContactPairs()) {
+      double dt = 0.0;
+      contact_.update(0, 0.0, dt);
+    }
+
     if (is_quasistatic_) {
       residual_with_bcs_ = buildQuasistaticOperator();
 
       // the residual calculation uses the old stiffness matrix
       // to help apply essential boundary conditions, so we
       // compute J here to prime the pump for the first solve
-      residual_with_bcs_->GetGradient(displacement_);
+      // TODO (EBC): looks like this is done when we call quasiStaticSolve()?
```

Comment:
Yes - I think we can delete this since you moved it to that call. I'll check the logic to be sure...